canvas/gradient.go

```diff
@@ -73,6 +73,15 @@ func (g *LinearGradient) Move(pos fyne.Position) {
 	repaint(g)
 }
 
+func (g *LinearGradient) Resize(size fyne.Size) {
+	if size == g.Size() {
+		return
+	}
+	g.baseObject.Resize(size)
+
+	g.Refresh()
```

Comment:
I would just add a quick comment here (and in RadialGradient) saying that the reason a Refresh is needed is that the texture will need to be regenerated. It's not a common pattern to Refresh in a resize (but it makes total sense here), so let's make sure someone wouldn't remove it later thinking they're making an optimization

Reply:
Added a small comment to both functions, to have more context. I expect people trying to delete existing code to be good friends with git history, but that's not always the case :)